src/io/serializer.hpp

```diff
@@ -18,6 +19,11 @@ class Serializer {
     virtual void* GetContext() { return this; }
     virtual PetscSerializeFunction GetSerializeFunction() = 0;
     virtual void RestoreTS(TS ts) = 0;
+
```

Comment:
Can you add some doxygen comment to this to explain why it is here.  Also, I think that we may wan to use the operator overloading ()